crates/ra_hir/src/source_binder.rs

```diff
@@ -216,7 +216,14 @@ impl SourceAnalyzer {
     }
 
     pub fn type_of(&self, db: &impl HirDatabase, expr: &ast::Expr) -> Option<Type> {
-        let expr_id = self.expr_id(expr)?;
+        let expr_id = if let Some(macro_call) = ast::MacroCall::cast(expr.syntax().clone()) {
+            let mut expander = Expander::new(db, self.file_id, self.body_owner?.module(db).id);
```

Comment:
Hm, why do we need to call the Expander manually here, instead of just using `parse_or_expand` on the corresponding macro file?